### ISSUE TITLE
💅 Shows "All these imports are only used as types" in Vue+typescript

### ISSUE THREAD

**Issue by lethefrost** [biomejs/biome#4539]
### Environment information

```bash
Biome 1.9.4(bundled)
```

### Rule name

useImportType

### Playground link

https://biomejs.dev/playground/?files.main.vue=PAB0AGUAbQBwAGwAYQB0AGUAPgAKACAAIAA8AFQAZQBzAHQAQwBvAG0AcABvAG4AZQBuAHQAPgAKACAAIAAgACAALgAuAC4ACgAgACAAPAAvAFQAZQBzAHQAQwBvAG0AcABvAG4AZQBuAHQAPgAKADwALwB0AGUAbQBwAGwAYQB0AGUAPgAKAAoAPABzAGMAcgBpAHAAdAAgAHMAZQB0AHUAcAAgAGwAYQBuAGcAPQAiAHQAcwAiAD4ACgAgACAAIAAgAGkAbQBwAG8AcgB0ACAAVABlAHMAdABDAG8AbQBwAG8AbgBlAG4AdAAgAGYAcgBvAG0AIAAnAC4ALwBUAGUAcwB0AEMAbwBtAHAAbwBuAGUAbgB0AC4AdgB1AGUAJwAKACAAIAAgACAAYwBvAG4AcwB0ACAAdABlAHMAdABUAHkAcABlACAAPQAgAEkAbgBzAHQAYQBuAGMAZQBUAHkAcABlADwAdAB5AHAAZQBvAGYAIABUAGUAcwB0AEMAbwBtAHAAbwBuAGUAbgB0AD4ACgA8AC8AcwBjAHIAaQBwAHQAPgA%3D

### Expected result

Please see the playground: if the line of `typeof` doesn't exist, biome won't have problem with unused import or something, but when you have the `typeof` line and when it is the only reference to the imported component inside the `script` tag, biome will report `All these imports are only used as types.` despite the usage outside the `script` tag.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
We only partially support vue files right now. See: https://biomejs.dev/internals/language-support/#html-super-languages-support

We recommend disabling these rules for these files until they are better supported.

```json
  {
    "overrides": [
      {
        "include": ["*.svelte", "*.astro", "*.vue"],
        "linter": {
          "rules": {
            "style": {
              "useConst": "off",
              "useImportType": "off"
            }
          }
        }
      }
    ]
  }
```
